components/list/list.js

```diff
@@ -437,9 +446,21 @@ class List extends PageableMixin(SelectionMixin(LitElement)) {
 
 	_handleSlotChange() {
 		this._updateItemShowingCount();
-		this.getItems().forEach((item, i) => {
-			if (i === 0) item.first = true;
-			else item.first = false;
+		const items = this.getItems();
```

Comment:
I suspect it was probably called out previously since it was already being done for `first`, but I really wish we weren't mutating the consumer's DOM, as it goes against the principle of not mutating DOM outside of the owner's `render`.